tests/v2/validation/clustertemplates/provisioning_clustertemplates.go

```diff
@@ -0,0 +1,96 @@
+package clustertemplates
+
+import (
+	"github.com/rancher/rancher/tests/v2/actions/clusters"
+	"github.com/rancher/rancher/tests/v2/actions/clustertemplates"
+	"github.com/rancher/rancher/tests/v2/actions/provisioning"
+	"github.com/rancher/rancher/tests/v2/actions/provisioninginput"
+	"github.com/rancher/shepherd/clients/rancher"
+	mgmt "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	"github.com/rancher/shepherd/pkg/namegenerator"
+)
+
+const (
+	CniCalico                        = "calico"
+	ClusterEnforcementSetting        = "cluster-template-enforcement"
+	EnabledClusterEnforcementSetting = "true"
+	IsRequiredQuestion               = true
+	UserPrincipalID                  = "local://"
+)
+
+var (
+	Questions = []mgmt.Question{
+		0: {
+			Variable: "rancherKubernetesEngineConfig.kubernetesVersion",
+			Default:  "",
+			Required: IsRequiredQuestion,
+			Type:     "string",
+		},
+		1: {
+			Variable: "rancherKubernetesEngineConfig.network.plugin",
+			Default:  "",
+			Required: IsRequiredQuestion,
+			Type:     "string",
+		},
+		2: {
+			Variable: "rancherKubernetesEngineConfig.services.etcd.backupConfig.intervalHours",
+			Default:  "",
+			Required: IsRequiredQuestion,
+			Type:     "string",
+		},
+	}
+)
+
+func CreateRkeTemplate(client *rancher.Client, members []mgmt.Member) (*mgmt.ClusterTemplate, error) {
+	rkeTemplateName := mgmt.ClusterTemplate{
+		Name:    namegenerator.AppendRandomString("rketemplate-"),
+		Members: members,
+	}
+
+	createTemplate, err := client.Management.ClusterTemplate.Create(&rkeTemplateName)
+	return createTemplate, err
+}
+
+func CreateRkeTemplateRevision(client *rancher.Client, templateRevisionConfig mgmt.ClusterTemplateRevision, templateId string) (*mgmt.ClusterTemplateRevision, error) {
+	rkeTemplateConfig := clustertemplates.NewRKE1ClusterTemplateRevisionTemplate(templateRevisionConfig, templateId)
+
+	clusterTemplateRevision, err := client.Management.ClusterTemplateRevision.Create(&rkeTemplateConfig)
+	if err != nil {
+		return nil, err
+	}
+	return clusterTemplateRevision, nil
+}
+
+func UpdateClusterTemplateClusterConfig(templateID, revisionID string, answers *mgmt.Answer) *mgmt.Cluster {
+
+	return &mgmt.Cluster{
+		DockerRootDir:                 "/var/lib/docker",
+		Name:                          namegenerator.AppendRandomString("rketemplate-cluster-"),
+		ClusterTemplateID:             templateID,
+		ClusterTemplateRevisionID:     revisionID,
+		ClusterTemplateAnswers:        answers,
+		RancherKubernetesEngineConfig: nil,
+	}
+}
+
+func CreateRke1ClusterWithTemplateAndVerify(client *rancher.Client, rke1Cluster *mgmt.Cluster, provisioningConfig *provisioninginput.Config) (*mgmt.Cluster, *clusters.ClusterConfig, error) {
```

Comment:
can you clarify in the godoc this verifies the clusterTemplate, not the created cluster